src/main/java/com/larscheidschmitzhermes/nexus3/github/oauth/plugin/api/GithubApiClient.java

```diff
@@ -95,47 +93,69 @@ private GithubPrincipal doAuthz(String loginName, char[] token) throws GithubAut
     }
 
     private GithubUser retrieveGithubUser(String loginName, char[] token) throws GithubAuthenticationException {
-        try {
-            HttpGet userRequest = new HttpGet(configuration.getGithubUserUri());
-            userRequest.addHeader(constructGithubAuthorizationHeader(token));
-            HttpResponse userResponse = client.execute(userRequest);
+        try (InputStreamReader reader = executeGet(configuration.getGithubUserUri(), token)) {
 
-            if (userResponse.getStatusLine().getStatusCode() != 200) {
-                LOGGER.warn("Authentication failed, status code was {}",
-                        userResponse.getStatusLine().getStatusCode());
-                userRequest.releaseConnection();
-                throw new GithubAuthenticationException("Authentication failed.");
-            }
-
-            GithubUser githubUser = mapper.readValue(new InputStreamReader(userResponse.getEntity().getContent()), GithubUser.class);
+            GithubUser githubUser = mapper.readValue(reader, GithubUser.class);
 
-            if (!loginName.equals(githubUser.getLogin())){
+            if (!loginName.equals(githubUser.getLogin())) {
                 throw new GithubAuthenticationException("Given username does not match Github Username!");
             }
 
+            if (configuration.getGithubOrg() != null && configuration.getGithubOrg().equals("")) {
+                checkUserInOrg(configuration.getGithubOrg(), token);
+            }
             return githubUser;
+
         } catch (IOException e) {
             throw new GithubAuthenticationException(e);
         }
     }
 
-    private Set<String> generateRolesFromGithubOrgMemberships(char[] token) throws GithubAuthenticationException{
-        HttpGet teamsRequest = new HttpGet(configuration.getGithubUserTeamsUri());
-        teamsRequest.addHeader(constructGithubAuthorizationHeader(token));
-        HttpResponse teamsResponse;
+    private void checkUserInOrg(String githubOrg, char[] token) throws GithubAuthenticationException {
+        Set<GithubOrg> orgs;
+        try (InputStreamReader reader = executeGet(configuration.getGithubUserOrgsUri(), token)) {
+            orgs = mapper.readValue(reader, new TypeReference<Set<GithubOrg>>() {
+            });
+        } catch (IOException e) {
```

Comment:
why not move essentially this try / catch block (= the mapping) to a separate method that already throws a `GithubAuthenticationException` ? That would make this method even more focused on the business logic!